src/inc/Admin/EmailLogsTab.php

```diff
@@ -121,17 +122,61 @@ public function process_email_content_preview() {
             return;
         }
 
+        $allowed_capability = WPML_Init::getInstance()->getService( 'plugin' )->getSetting( 'can-see-submission-data', 'manage_options' );
+
+        if ( ! current_user_can( $allowed_capability ) ) {
+            return;
+        }
+
         // Get the mail log.
         $mail = Mail::find_one( absint( $_GET['email_log_id'] ) );
 
         if ( ! $mail ) {
             return;
         }
 
-        echo $mail->get_message();
+        echo $this->get_html_preview_message( $mail->get_message() );
         exit;
     }
 
+    /**
+     * Get the message to be rendered in the preview.
+     *
+     * @since {VERSION}
+     *
+     * @param string $message Email log message.
+     *
+     * @return string
+     */
+    private function get_html_preview_message( $message ) {
+
+        // Loosely test if the message is HTML.
+        if ( preg_match( '/<[a-z][^\s>\/]*>/i', $message ) ) {
```

Comment:
This regex will miss a lot of HTML, examples: <CAPITALS>, <tags with spaces>, which create possible bypasses around the next step. 

It may be a complicated task to accurately detect if the message includes HTML. If the purpose is only to convert <script> into an escaped version of <script> then it may not be worthwhile for a fringe edge case.

Reply:
Thank you for your comment @robertrofam. I've pushed the regex change with 8ba2b233fa31f12def5bd4c98e2cba023da48f81. 

How about the `wp_kses()`? Do you think that escaping `<script>` tags is enough?